openapi.go

```diff
@@ -79,6 +79,28 @@ func NewOpenApiSpec() openapi3.T {
 	return spec
 }
 
+type OpenAPIServable interface {
+	SpecHandler()
```

Comment:
Need to think of way to make this contract a bit more binding? 🤔 

Maybe not. The whole point is that people can server these whatever way they want. We're basically just dealing with Engine config for them.

Reply:
Yeah exactly, most people won't even see anything of this. But I'll feed the engine in the method parameters anyway, because we use the `OpenAPIServable` only inside `RegisterOpenAPIRoutes` which ALSO depends on the `Engine`, and this allows a clean syntax for alternative routers

https://github.com/go-fuego/fuego/pull/344#discussion_r1922446176